### PR TITLE
Yy feature create react native tab view for profile pafe

### PR DESCRIPTION
**Pull request by yenyin262**
-implement heartIcon & tabview functionality on profile page together with styling

### CODE REVIEW

File: client/js/components/TabView/TabView.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import React, {Component} from 'react';
+import {View, Text, StyleSheet, Image} from 'react-native';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import Browse from '../../screens/Browse';
+import Faves from '../Faves';
+import styles from './styles';
+
+class TabViewProfile extends Component {
+  constructor() {
+    super();
```
Comment (siddharthparmar7) [inline client/js/components/TabView/TabView.js]: pass props arg in your constructor and super.

File: client/js/components/TabView/TabView.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  onChangeStyle(tab) {
+    this.setState({
+      items: false,
+      store: false,
+      brands: false,
+      myphotos: false,
+    });
+    this.setState({[tab]: true});
```
Comment (siddharthparmar7) [inline client/js/components/TabView/TabView.js]: This should not work. How are you resetting a tab from true to false?

File: client/js/screens/Profile/Profile.js [hunk 3]
```diff
@@ -1,23 +1,46 @@
 import React from 'react';
-import { View, Text, Button } from 'react-native';
-import { createToken, deleteToken, queryToken } from "../../config/models/authentication"
-import { withNavigation } from 'react-navigation';
+import {View, Text, Button, Image, TouchableOpacity} from 'react-native';
+import TabViewProfile from '../../components/TabView';
+import {withNavigation} from 'react-navigation';
+import styles from './styles';
+// import Icon from 'react-native-vector-icons/Ionicons';
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js]: remove commented code.

File: client/js/screens/Profile/Profile.js [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+  createToken,
+  deleteToken,
+  queryToken,
+} from '../../config/models/authentication';
+
 import AsyncStorage from '@react-native-community/async-storage';
-const Profile = ({ navigation, user }) => {
-  console.log(user)
+const Profile = ({navigation, user}) => {
+  console.log(user);
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js]: remove console log.

File: client/js/screens/Profile/Profile.js [hunk 5]
```diff
@@ -1,23 +1,46 @@
...
+
   return (
-    <View>
+    <View style={{flex: 1}}>
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js]: move your styles to styles.js.

File: client/js/screens/Profile/Profile.js [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
-
-        }
+        title={'Log Out'}
+        onPress={async () => {
+          await deleteToken();
+          navigation.navigate('AuthLoading');
+        }}
       />
-      <Text>Profile Page</Text>
+      <View style={{flexDirection: 'row', marginHorizontal: '3.5%'}}>
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js]: move your styles to styles.js.

File: client/js/screens/Profile/Profile.js [hunk 7]
```diff
@@ -1,23 +1,46 @@
...
+            }}>
+            <Text style={styles.editProfile}>Edit Profile</Text>
+          </TouchableOpacity>
+        </View>
+      </View>
+
+      <TabViewProfile navigation={navigation} />
     </View>
   );
 };
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js:50]: move all your inline styles to styles.js.

File: client/package.json [hunk 8]
```diff
@@ -22,6 +21,7 @@
     "react-native-check-box": "^2.1.7",
     "react-native-gesture-handler": "^1.4.1",
     "react-native-reanimated": "^1.2.0",
+    "react-native-tab-view": "^2.10.0",
```
Comment (siddharthparmar7) [inline client/package.json]: New packages should be installed by your team leader. Please make the changes.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please resolve merge conflicts and make the requested changes. Thank you.

File: client/js/components/TabView/TabView.js (hunk 2 above)
Comment (zareef4) [reply client/js/components/TabView/TabView.js]: I think shes actually resetting the whole state on every change... and then uopdating the tab to be true

File: client/js/components/TabView/TabView.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  }
+  render() {
+    return (
+      <View>
```
Comment (zareef4) [inline client/js/components/TabView/TabView.js]: I think if you have several repeated code blocks you shold either:

1. make it a reusable component. Do this if you can see potential uses of this compenent in other screens etc.
2. use a map and render the repeated block for each of the changing data (in this case the tab name)

File: client/js/components/TabView/index.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+import TabViewProfile from './TabView';
```
Comment (zareef4) [inline client/js/components/TabView/index.js]: empty component?

File: client/js/components/faves/index.js [hunk 11]
```diff
@@ -0,0 +1,2 @@
+import Faves from './Faves';
```
Comment (zareef4) [inline client/js/components/faves/index.js:1]: empty component again

File: client/js/components/faves/faves.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+      {faveBrandIds.length > 0 ? (
+        <Brand brands={brands} faveBrandIds={faveBrandIds} />
+      ) : (
+        <View style={styles.container}>
+          <Text style={styles.favesText}> You have no favorites. </Text>
+        </View>
+      )}
+
+      {/* {faveStoreIds.length > 0 ? (
```
Comment (iyuvraajsingh) [inline client/js/components/faves/faves.js]: Please remove the commented code.

File: client/js/screens/Profile/styles.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  container: {
+    flex: 1,
+  },
+  imgWrapper: {
+    flexDirection: 'row',
+    marginHorizontal: '3.5%',
+  },
+  btnWrapper: {
+    marginTop: 20,
```
Comment (iyuvraajsingh) [inline client/js/screens/Profile/styles.js:32]: Please use %s if possible.

File: client/js/screens/Profile/styles.js [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    fontSize: 16,
+    textDecorationLine: 'underline',
+  },
+  img: {
+    width: 60,
+    height: 60,
+    borderRadius: 30,
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    marginVertical: 20,
```
Comment (iyuvraajsingh) [inline client/js/screens/Profile/styles.js:19]: Please use %s if possible.

File: client/js/components/TabView/TabView.js [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+                    : require('../../assets/inactiveheart.png')
+                }
+              />
+              <TouchableOpacity onPress={() => this.onChangeStyle('myphotos')}>
+                <Text>My Photos</Text>
+              </TouchableOpacity>
+            </View>
+          </View>
+        </View>
+        {/* {this.state.items && <Browse />} */}
```
Comment (iyuvraajsingh) [inline client/js/components/TabView/TabView.js]: Please remove the commented code.

**Review by iyuvraajsingh — CHANGES_REQUESTED** [review]

File: client/js/components/Faves/FavesContainer.js [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+          removeFaveItem,
+          removeFaveBrand,
+        }) => (
+          <Query query={GET_All_ITEMS}>
+            {({loading, error, data}) => {
+              if (loading) return <Loader />;
+              if (error) return <Text>{error.message}</Text>;
+              const itemData = data.allItems;
+              console.log(faveItemIds, 'faveitem');
+              // console.log(faveBrandIds, 'brand');
```
Comment (siddharthparmar7) [inline client/js/components/Faves/FavesContainer.js]: Please remove all the console logs from your branch.

File: client/js/components/Faves/FavesContainer.js [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+              );
+            }}
+          </Query>
+        )}
+      </FavesItemsContext.Consumer>
+    );
+  }
+}
+
+export default withNavigation(FavesContainer);
```
Comment (siddharthparmar7) [inline client/js/components/Faves/FavesContainer.js:143]: Please remove all the comments and console logs.

File: client/js/components/TabViewProfile/TabViewProfile.js [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+              <TouchableOpacity onPress={() => this.onChangeStyle('myphotos')}>
+                <Text>My Photos</Text>
+              </TouchableOpacity>
+            </View>
+          </View>
+        </View>
+
+        {/* {this.state.items && <FavesContainer items={items} />}
+        {this.state.store && <FavesContainer store={store} />}
+        {this.state.brands && <FavesContainer brands={brands} />} */}
```
Comment (siddharthparmar7) [inline client/js/components/TabViewProfile/TabViewProfile.js]: remove these comments.

File: client/js/components/faves/faves.js [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+      ) : (
+        <View style={styles.container}>
+          <Text style={styles.favesText}> You have no favorites. </Text>
+        </View>
+      )} */}
+    </ScrollView>
+  );
+};
+
+export default Faves;
```
Comment (siddharthparmar7) [inline client/js/components/faves/faves.js:135]: Please remove all the unnecessary comments.

File: client/js/context/FaveItemsContext/FaveItemsContext.js [hunk 20]
```diff
@@ -13,30 +17,31 @@ class FaveItemsProvider extends Component {
     try {
       const savedFaveItems = await queryFaveItems();
       const faveItemIds = savedFaveItems.map(faveItem => faveItem);
-      this.setState({ faveItemIds })
+      this.setState({faveItemIds});
+      console.log(this.state.faveItemIds);
```
Comment (siddharthparmar7) [inline client/js/context/FaveItemsContext/FaveItemsContext.js]: remove console log.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]

File: client/js/screens/Profile/Profile.js (hunk 6 above)
Comment (siddharthparmar7) [reply client/js/screens/Profile/Profile.js]: Please move your styles to styles.js file.

File: client/js/screens/Profile/Profile.js [hunk 21]
```diff
@@ -1,27 +1,47 @@
...
+      </View>
+      {/* <TouchableOpacity
+        onPress={() => {
+          Linking.openURL('http://maps.apple.com/?ll=37.484847,-122.148386');
+        }}>
         <Text>Open Map</Text>
-      </TouchableOpacity>
+      </TouchableOpacity> */}
+      <TabViewProfile />
     </View>
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js:48]: remove commented code if not needed.

File: client/js/components/faves/faves.js [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+            <TouchableOpacity
+              onPress={() => navigation.navigate('Stores&Brands')}>
+              <Text style={styles.browseTxt}>Browse Stores</Text>
+            </TouchableOpacity>
+          </View>
+        ))}
+    </ScrollView>
+  );
+};
+
```
Comment (siddharthparmar7) [inline client/js/components/faves/faves.js:134]: Please remove all the autogenerated ```{' '}```

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]

**Review by siddharthparmar7 — APPROVED** [review]
Nice work!

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please resolve merge conflicts so I can merge your branch into develop. Thank you.

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**